tpm2/src/tpm/storage_key_tools.c

```diff
@@ -23,12 +23,48 @@
 int get_srk_handle(TSS2_SYS_CONTEXT * sapi_ctx,
                    TPM2_HANDLE * srk_handle,
                    TPM2B_AUTH * storage_hierarchy_auth)
+{
+  // Get the list of objects in TPM persistent storage and check them all
+  // against the SRK criteria. Upon return from get_existing_srk_handle(),
+  // the srk_handle parameter passed to get_existing_srk_handle() is either
+  // zero (none of the loaded objects is the SRK) or contains the handle used
+  // to reference the SRK in TPM persistent storage. The function
+  // get_existing_srk_handle() will set next_persistent_handle to the next
+  // available TPM persistent storage location where the SRK can be put.
+  TPM2_HANDLE next_persistent_handle = 0;
+
+  get_existing_srk_handle(sapi_ctx, srk_handle, &next_persistent_handle);
```

Comment:
This needs to check for error conditions, e.g. it returns 1 if sapi_ctx is null.